File: c/collatz-conjecture/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start) {
  if (start <= 0) {
    return ERROR_VALUE;
  }

  int steps = 0;
  while (start > 1) {
    start = start % 2 == 0 ? start / 2 : (start * 3) + 1;
    steps++;
  }

  return steps;
}

File: csharp/collatz-conjecture/CollatzConjecture.cs
using System;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        ArgumentOutOfRangeException.ThrowIfNegativeOrZero(number);

        var steps = 0;
        var n = number;
        while (n > 1)
        {
            n = n % 2 == 0 ? n / 2 : (3 * n) + 1;
            steps++;
        }
        return steps;
    }
}
